File: reports/20260826-143732/results/app-configuration/data-plane/python/crud/app-configuration-dp-python-crud/python-azure-tools/with-azure-tools/generated-code/app_configuration_crud.py
import logging
import os

from azure.appconfiguration import (
    AzureAppConfigurationClient,
    ConfigurationSetting,
    FeatureFlagConfigurationSetting,
)
from azure.core.exceptions import HttpResponseError
from azure.identity import DefaultAzureCredential


SETTING_KEY = "app:Settings:FontSize"
SETTING_VALUE = "24"
PRODUCTION_LABEL = "Production"
KEY_FILTER = "app:Settings:*"


def main() -> int:
    endpoint = os.environ.get("AZURE_APPCONFIGURATION_ENDPOINT")
    if not endpoint:
        raise RuntimeError(
            "Set AZURE_APPCONFIGURATION_ENDPOINT to your App Configuration endpoint."
        )

    credential = DefaultAzureCredential()
    try:
        with AzureAppConfigurationClient(
            base_url=endpoint,
            credential=credential,
        ) as client:
            client.set_configuration_setting(
                ConfigurationSetting(key=SETTING_KEY, value=SETTING_VALUE)
            )
            client.set_configuration_setting(
                ConfigurationSetting(
                    key=SETTING_KEY,
                    value=SETTING_VALUE,
                    label=PRODUCTION_LABEL,
                )
            )

            setting = client.get_configuration_setting(key=SETTING_KEY)
            print(f"{setting.key} = {setting.value}")

            print(f"Settings matching {KEY_FILTER!r}:")
            for matching_setting in client.list_configuration_settings(
                key_filter=KEY_FILTER
            ):
                label = matching_setting.label or "(no label)"
                print(
                    f"  {matching_setting.key} [{label}] = "
                    f"{matching_setting.value}"
                )

            client.set_configuration_setting(
                FeatureFlagConfigurationSetting(
                    feature_id="BetaFeature",
                    enabled=True,
                )
            )

            client.delete_configuration_setting(key=SETTING_KEY)
    except HttpResponseError as error:
        status_code = error.status_code or "unknown"
        logging.error(
            "Azure App Configuration request failed (HTTP %s): %s",
            status_code,
            error.message,
        )
        return 1
    finally:
        credential.close()

    return 0


if __name__ == "__main__":
    logging.basicConfig(level=logging.INFO, format="%(levelname)s: %(message)s")
    raise SystemExit(main())

File: reports/20260826-202215/results/app-configuration/data-plane/java/crud/app-configuration-dp-java-crud/java-azure-tools/with-azure-tools/generated-code/src/main/java/com/example/AppConfigurationCrud.java
package com.example;

import com.azure.core.exception.HttpResponseException;
import com.azure.data.appconfiguration.ConfigurationClient;
import com.azure.data.appconfiguration.ConfigurationClientBuilder;
import com.azure.data.appconfiguration.models.ConfigurationSetting;
import com.azure.data.appconfiguration.models.FeatureFlagConfigurationSetting;
import com.azure.data.appconfiguration.models.SettingSelector;

public final class AppConfigurationCrud {
    private static final String CONNECTION_STRING_ENV = "AZURE_APPCONFIG_CONNECTION_STRING";
    private static final String SETTING_KEY = "app:Settings:FontSize";
    private static final String PRODUCTION_LABEL = "Production";

    private AppConfigurationCrud() {
    }

    public static void main(String[] args) {
        String connectionString = System.getenv(CONNECTION_STRING_ENV);
        if (connectionString == null || connectionString.isBlank()) {
            System.err.printf("Set the %s environment variable before running.%n", CONNECTION_STRING_ENV);
            System.exit(1);
        }

        ConfigurationClient client = new ConfigurationClientBuilder()
            .connectionString(connectionString)
            .buildClient();

        try {
            client.setConfigurationSetting(SETTING_KEY, null, "24");
            client.setConfigurationSetting(SETTING_KEY, PRODUCTION_LABEL, "24");

            ConfigurationSetting setting = client.getConfigurationSetting(SETTING_KEY, null);
            System.out.printf("%s = %s%n", setting.getKey(), setting.getValue());

            SettingSelector selector = new SettingSelector()
                .setKeyFilter("app:Settings:*");
            client.listConfigurationSettings(selector)
                .forEach(item -> System.out.printf(
                    "%s [%s] = %s%n",
                    item.getKey(),
                    item.getLabel() == null ? "no label" : item.getLabel(),
                    item.getValue()));

            FeatureFlagConfigurationSetting betaFeature =
                new FeatureFlagConfigurationSetting("BetaFeature", true);
            client.setConfigurationSetting(betaFeature);

            client.deleteConfigurationSetting(SETTING_KEY, null);
            System.out.printf("Deleted setting: %s%n", SETTING_KEY);
        } catch (HttpResponseException exception) {
            int statusCode = exception.getResponse() == null
                ? -1
                : exception.getResponse().getStatusCode();
            System.err.printf(
                "Azure App Configuration request failed (HTTP %s): %s%n",
                statusCode == -1 ? "unknown" : Integer.toString(statusCode),
                exception.getMessage());
            System.exit(1);
        }
    }
}
